include/libcyphal/transport/udp/tx_rx_sockets.hpp

```diff
@@ -96,6 +101,27 @@ class IRxSocket
     IRxSocket& operator=(const IRxSocket&)     = delete;
     IRxSocket& operator=(IRxSocket&&) noexcept = delete;
 
+    /// @brief Defines the error types that may occur during the `receive` operation.
+    ///
+    using ReceiveFailure = cetl::variant<PlatformError, ArgumentError, MemoryError>;
+
+    struct ReceiveSuccess
+    {
+        TimePoint   timestamp;
+        std::size_t payload_size;
+        // TODO: Use CETL pmr instead of std::unique_ptr.
+        std::unique_ptr<cetl::byte[]> payload;  // NOLINT(*-avoid-c-arrays)
```

Comment:
This is very temporary solution (and of course it's heap based - hence the `// TODO`). I need figure out how better support array allocations at our CETL pmr (it seems that currently it's not yet supported by the CETL).